ion-schema/src/system.rs

```diff
@@ -931,30 +1039,57 @@ impl Resolver {
                     _ => Err(error),
                 },
                 Ok(schema_content) => {
-                    let isl = self.isl_schema_from_elements(schema_content.into_iter(), id)?;
-                    self.schema_from_isl_schema(isl, id, type_store, load_isl_import)
+                    // ISL version marker regex
+                    let isl_version_marker = Regex::new(r"^\$ion_schema_\d.*$").unwrap();
+                    let mut isl_version = IonSchemaLanguageVersion::V1_0;
+
+                    // find the ISL version
+                    for value in &schema_content {
+                        // verify if value is an ISL version marker and if it has valid format
+                        if value.ion_type() == IonType::Symbol
```

Comment:
If `value` is an Ion struct that is annotated with `type::` or `schema_header::` (and no version marker has been encountered yet), then we know it's ISL 1.0, and we can break out of the loop.